src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/ReportingConfiguration.cs

```diff
@@ -168,41 +186,67 @@ public async ValueTask<ScenarioRun> CreateScenarioRunAsync(
         string scenarioName,
         string iterationName = Defaults.DefaultIterationName,
         IEnumerable<string>? additionalCachingKeys = null,
+        IEnumerable<string>? additionalTags = null,
         CancellationToken cancellationToken = default)
     {
         ChatConfiguration? chatConfiguration = ChatConfiguration;
+        ChatDetails? chatDetails = null;
 
-        if (chatConfiguration is not null && ResponseCacheProvider is not null)
+        IEnumerable<string>? tags;
+        if (additionalTags is null)
+        {
+            tags = Tags;
+        }
+        else if (Tags is null)
+        {
+            tags = additionalTags;
+        }
+        else
+        {
+            tags = [.. Tags, .. additionalTags];
+        }
```

Comment:
This could be simplified to

`tags = [.. Tags ?? [], .. additionalTags ?? []];`

Reply:
Ah I was trying to get `tags` to be `null` when there are no tags. The change proposed above would create an empty collection instead (and at that point the `Tags` property above as well as `ScenarioRunResults.Tags` could also be non-nullable).

This is one of the tensions I was describing earlier where with collections deciding between empty v/s `null` for optional stuff can be tricky. In this case, I favored `null` (both on `ReportingConfiguration` and `ScenarioRunResult`) to avoid any impact on serialized data when tags are not being used.

`CachingKeys` above is in a similar boat - however, since we always add some default keys (from the `IChatClient` metadata), we favored a non-nullable property for that.